lib/ansible/plugins/callback/default.py

```diff
@@ -115,6 +115,9 @@ def v2_on_file_diff(self, result):
         if 'diff' in result._result and result._result['diff']:
             self._display.display(self._get_diff(result._result['diff']))
 
+    def _get_censored_item(self, result):
```

Comment:
I'll check, but i believe callbacks already get a copy, just need to check that it is not linked to registered vars, we should be able to censor at that point.